tests/unit/s2n_resume_test.c

```diff
@@ -0,0 +1,248 @@
+/*
+ * Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License").
+ * You may not use this file except in compliance with the License.
+ * A copy of the License is located at
+ *
+ *  http://aws.amazon.com/apache2.0
+ *
+ * or in the "license" file accompanying this file. This file is distributed
+ * on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+ * express or implied. See the License for the specific language governing
+ * permissions and limitations under the License.
+ */
+
+#include "s2n_test.h"
+
+#include "tls/s2n_tls.h"
+#include "utils/s2n_safety.h"
+#include "tls/s2n_resume.h"
+/* To test static function */
+#include "tls/s2n_resume.c"
+
+#define S2N_TLS13_STATE_SIZE_WITHOUT_SECRET    S2N_MAX_STATE_SIZE_IN_BYTES - S2N_TLS_SECRET_LEN
```

Comment:
Nit: extra space between `S2N_TLS13_STATE_SIZE_WITHOUT_SECRET` and `S2N_MAX_STATE_SIZE_IN_BYTES`